lib/src/message.dart

```diff
@@ -31,7 +31,7 @@ abstract class Message {
   /// by concatenating them with a comma.
   ///
   /// The returned map is unmodifiable.
-  Map<String, String> get headers => _headers.singleValues;
+  Map<String, String?> get headers => _headers.singleValues;
```

Comment:
Would `Map<String, String?>` -> `Map<String, String>` break a lot of things in shelf?
(I guess I'd have to clone and play with to really see it).

side note: I kind of wish we deprecated `Message.headers` in favor of `Message.headersAll`, since the comma-separate-value is rather surprising.

Reply:
> I don't see any path where null headers were ever really existing in practice.

I agree, indeed `Headers.singleValues` has the type `Map<String, String>`, so the value here cannot (if you believe in null-safety) be `null`.

> `headersAll` is annoying because most of the time, you just want one value

I suppose that's true, and the failure scenario isn't super bad.

Reply:
So...any other feedback?

Reply:
I am not following why  this is not a `Map<String, String>`?

Reply:
Oh wait...it should be.

Weird...will look

Reply:
just a mistake on my part!

Reply:
fixed